src/metaswitch/clearwater/queue_manager/alarms.py

```diff
@@ -0,0 +1,65 @@
+# Project Clearwater - IMS in the Cloud
+# Copyright (C) 2015 Metaswitch Networks Ltd
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+from metaswitch.common.alarms import issue_alarm as int_issue_alarm
+import logging
+
+_log = logging.getLogger("queue_manager.alarms")
+
+def issue_alarm(identifier): # pragma : no cover
+    int_issue_alarm("queue-manager", identifier)
+
+class QueueAlarm(object):
+    def __init__(self, clear, minor, critical, name):
+        self._clear_alarm = clear
+        self._minor_alarm = minor
+        self._critical_alarm = critical
+        self._name = name
+        self._current_alarm = clear
```

Comment:
I'm not sure we need to track the current alarm - the alarms agent has functionality to suppress alarms that are re-raised too frequently, and re-raising may be useful (e.g. recovery after an alarm agent outage).